app/app.go

```diff
@@ -9,276 +9,255 @@ import (
 	"github.com/cosmos/cosmos-sdk/simapp"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/cosmos/cosmos-sdk/types/module"
+	"github.com/cosmos/cosmos-sdk/version"
 	"github.com/cosmos/cosmos-sdk/x/auth"
+	"github.com/cosmos/cosmos-sdk/x/auth/ante"
+	"github.com/cosmos/cosmos-sdk/x/auth/vesting"
 	"github.com/cosmos/cosmos-sdk/x/bank"
+	"github.com/cosmos/cosmos-sdk/x/crisis"
+	distr "github.com/cosmos/cosmos-sdk/x/distribution"
+	"github.com/cosmos/cosmos-sdk/x/evidence"
 	"github.com/cosmos/cosmos-sdk/x/genutil"
+	"github.com/cosmos/cosmos-sdk/x/gov"
 	"github.com/cosmos/cosmos-sdk/x/mint"
-
 	"github.com/cosmos/cosmos-sdk/x/params"
+	paramsclient "github.com/cosmos/cosmos-sdk/x/params/client"
+	"github.com/cosmos/cosmos-sdk/x/slashing"
 	"github.com/cosmos/cosmos-sdk/x/staking"
+	"github.com/cosmos/cosmos-sdk/x/supply"
 
 	"github.com/ovrclk/akash/x/deployment"
 	"github.com/ovrclk/akash/x/market"
 	"github.com/ovrclk/akash/x/provider"
 
+	abci "github.com/tendermint/tendermint/abci/types"
 	"github.com/tendermint/tendermint/libs/log"
+	tmos "github.com/tendermint/tendermint/libs/os"
 	tmtypes "github.com/tendermint/tendermint/types"
 	dbm "github.com/tendermint/tm-db"
-
-	abci "github.com/tendermint/tendermint/abci/types"
-	tmos "github.com/tendermint/tendermint/libs/os"
-
-	"github.com/cosmos/cosmos-sdk/version"
-	distr "github.com/cosmos/cosmos-sdk/x/distribution"
-	"github.com/cosmos/cosmos-sdk/x/slashing"
-	"github.com/cosmos/cosmos-sdk/x/supply"
 )
 
 const (
 	appName = "akash"
 )
 
 var (
-	mbasics = module.NewBasicManager(
-		genutil.AppModuleBasic{},
 
-		// accounts, fees.
+	// ModuleBasics defines the module BasicManager is in charge of setting up basic,
+	// non-dependant module elements, such as codec registration
+	// and genesis verification.
+	ModuleBasics = module.NewBasicManager(
 		auth.AppModuleBasic{},
-
-		// tokens, token balance.
-		bank.AppModuleBasic{},
-
-		// total supply of the chain
 		supply.AppModuleBasic{},
-
-		// inflation
-		mint.AppModuleBasic{},
-
+		genutil.AppModuleBasic{},
+		bank.AppModuleBasic{},
 		staking.AppModuleBasic{},
-
-		slashing.AppModuleBasic{},
-
+		mint.AppModuleBasic{},
 		distr.AppModuleBasic{},
-
+		gov.NewAppModuleBasic(
+			paramsclient.ProposalHandler, distr.ProposalHandler,
+		),
 		params.AppModuleBasic{},
+		crisis.AppModuleBasic{},
+		slashing.AppModuleBasic{},
+		evidence.AppModuleBasic{},
 
 		// akash
 		deployment.AppModuleBasic{},
 		market.AppModuleBasic{},
 		provider.AppModuleBasic{},
 	)
+
+	// module account permissions
+	maccPerms = map[string][]string{
```

Comment:
the reason for having these be functions was to prevent mutation - the functions return a new copy every time.  please revert to the previous pattern.

Reply:
that's what `GetMaccPerms` does too 🙂

Reply:
why did you change it then?